projects/packages/forms/src/dashboard/inbox/dataviews/actions/index.js

```diff
@@ -0,0 +1,223 @@
+import { Icon } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import { seen, trash, backup } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { notSpam, spam } from '../../../icons';
+import { STORE_NAME } from '../../../state';
+import { ACTIONS } from '../../constants';
+import InboxResponse from '../../response';
+
+export const viewAction = {
+	id: 'view-response',
+	label: __( 'View response', 'jetpack-forms' ),
+	isPrimary: true,
+	icon: <Icon icon={ seen } />,
+	RenderModal: ( { items } ) => {
+		const [ item ] = items;
+		return <InboxResponse isLoading={ false } response={ item } />;
+	},
+};
+
+export const markAsSpamAction = {
+	id: 'mark-as-spam',
+	label: __( 'Mark as spam', 'jetpack-forms' ),
+	isEligible: item => item.status !== 'spam',
+	supportsBulk: true,
+	icon: <Icon icon={ spam } />,
```

Comment:
I had to do this in order to get the proper height/width in the `icon` (24). I'm not sure why this is happening though, as I didn't observe any differences from the `icons` in site editor lists with DataViews, regarding their consumption.

The difference though is that in the other lists probably [this path is returned](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/icon/index.tsx#L95), while here it didn't.

@ciampo @tyxla any thoughts about this?

Reply:
I'm not sure I understand, @ntsekouras, can you please elaborate on what the issue is?

Reply:
Sure. If you just add `icon: spam`, as core does, and not the `Icon` wrapper, the icons are bigger than the other lists in core. I couldn't understand why this is happening.

In core the height/width are smaller because the SVG gets the `height/width` attributes, probably by this [code path](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/icon/index.tsx#L95) and in this PR here it gets the `size` prop.

Makes sense?

Reply:
Can't we just pass a `size` prop? I've seen this done for both `Icon` and `Gridicon` quite extensively.

Reply:
Ideally, we wouldn't need the Icon wrapper at all, like in core.

Reply:
The core package [literally suggests](https://github.com/WordPress/gutenberg/tree/trunk/packages/icons#usage) using the `Icon` wrapper though:

```
import { Icon, check } from '@wordpress/icons';

<Icon icon={ check } />;
```

maybe I'm missing something?

Reply:
If I understand correctly, dataviews internally passes the `icon` to a `@wordpress/components` `Button` via its `icon` prop: https://github.com/WordPress/gutenberg/blob/75ef79ec8dc0ecefe4c3b180ca69d39a5445d02e/packages/dataviews/src/components/dataviews-item-actions/index.tsx#L77

In that case, I guess the `<Icon />` wrapper wouldn't be necessary, since the `Button` should be able to handle icons from the `@wordpress/icons` package directly.

If that's the case, maybe the answer is in how the `spam` icon is defined, compared to how `@wordpress/icons` are defined (especially in terms of viewbox and explicit width/height)

Reply:
Actually it's not only about the `spam` (introduced here), but other icons imported from `@wordpress/icons`.

Reply:
It's also fair to say that `Button` has both `icon` and `iconSize`, so the DataViews actions API might make sense to support `iconSize` as well, if we have a reasonable need for that.

Reply:
> Actually it's not only about the `spam` (introduced here), but other icons imported from `@wordpress/icons`.

We should investigate and understand what's the matter. My assumption is that they would play nicely out of the box.

> It's also fair to say that `Button` has both `icon` and `iconSize`, so the DataViews actions API might make sense to support `iconSize` as well, if we have a reasonable need for that.

My preference would be to keep the API layer as simple as possible.

Also, I'm not a fan of `Button`'s API approach, as it leads to an explosion of props. A better approach would be to allow for a render prop, so that the consumer can pass the `<Icon />` component and specify any other prop directly from consumer-land.